### PR TITLE
Redid the Action Popup

### PR DESCRIPTION
**Pull request by martholomew**
1. Changed the layout in favor of a more modern, material layout.
2. Added a dynamic tooltip indicating how to scan.
3. Redid some logic in the On/Off switch so that it updates when profiles are switched (there remains an edge case when the settings page is focused where it exhibits strange behavior).

### CODE REVIEW

File: ext/js/pages/action-popup-main.js [hunk 1]
```diff
@@ -81,6 +81,15 @@ class DisplayController {
 
     // Private
 
+    _updateDisplayModifierKey() {
```
Comment (jamesmaa) [inline ext/js/pages/action-popup-main.js:84]: Let's target the entire tooltip so we don't end up with "Hold no modifier to scan". In the case of no modifier lets have the string "hover over text to scan".

File: ext/js/pages/action-popup-main.js [hunk 2]
```diff
@@ -235,8 +247,8 @@ class DisplayController {
         /** @type {NodeListOf<HTMLSelectElement>} */
         const selects = document.querySelectorAll('.profile-select');
         /** @type {NodeListOf<HTMLElement>} */
-        const optionGroups = document.querySelectorAll('.profile-select-option-group');
-        for (let i = 0; i < Math.min(selects.length, optionGroups.length); i++) {
+        // const optionGroups = document.querySelectorAll('.profile-select-option-group');
```
Comment (jamesmaa) [inline ext/js/pages/action-popup-main.js:250]: Stray comment

File: ext/action-popup.html [hunk 3]
```diff
@@ -11,94 +11,64 @@
     <link rel="icon" type="image/png" href="/images/icon48.png" sizes="48x48">
     <link rel="icon" type="image/png" href="/images/icon64.png" sizes="64x64">
     <link rel="icon" type="image/png" href="/images/icon128.png" sizes="128x128">
+    <link rel="stylesheet" type="text/css" href="/css/material.css">
     <link rel="stylesheet" type="text/css" href="/css/action-popup.css">
     <script src="/js/pages/action-popup-main.js" type="module"></script>
 </head>
 <body>
-
-<div id="loading">
```
Comment (jamesmaa) [inline ext/action-popup.html:19]: Why is this removed?

File: ext/images/info.svg
Comment (jamesmaa) [inline ext/images/info.svg:1]: I kind of prefer the question mark over the info icon because I would imagine more users looking for help than looking for info

File: ext/action-popup.html [hunk 4]
```diff
@@ -11,94 +11,64 @@
...
+            <button type="button" class="low-emphasis action-open-permissions" title="Permissions" hidden>
+                <div class="action-icon">
+                    <span class="icon" data-icon="key"></span>
+                    <div class="nav-button-warning-badge permissions-required-warning" hidden>
+                        <div class="nav-button-warning-badge-outer"></div>
+                        <div class="nav-button-warning-badge-inner"></div>
+                    </div>
+                </div>
+            </button>
+            <button type="button" class="low-emphasis action-open-search" title="Search" data-hotkey='["global:openSearchPage","title","Search ({0})\nShift+click to open here"]'>
```
Comment (jamesmaa) [inline ext/action-popup.html:61]: I'm not seeing this title anymore

image

File: ext/css/action-popup.css [hunk 5]
```diff
@@ -497,6 +387,104 @@ select.profile-select {
...
+    padding: 10px;
+}
+
+.action-container button {
+    flex: 0 0 auto;
+}
+
+/* Misc */
+
+h2 {
```
Comment (jamesmaa) [inline ext/css/action-popup.css:468]: This is pretty broad. Can we scope it down more? Maybe we can use a span instead of an `h2` tag?

File: ext/action-popup.html [hunk 6]
```diff
@@ -11,94 +11,64 @@
...
-        <span class="link-group-icon" data-icon="magnifying-glass"></span><span class="link-group-label">Search</span>
-    </a>
-    <a tabindex="0" class="link-group action-open-info">
-        <span class="link-group-icon" data-icon="question-mark-circle"></span><span class="link-group-label">Information</span>
-    </a>
-</div>
-</div>
-
+        <div tabindex="0" class="action-container">
+            <button type="button" class="low-emphasis action-open-settings" title="Settings" data-hotkey='["global:openSettingsPage","title","Settings ({0})"]'>
```
Comment (jamesmaa) [inline ext/action-popup.html:47]: Idk what css element this is, but the blue border that shows up when you hover over the action icons shows up somewhat slow. Could you make the animation faster? If I move across the buttons somewhat fast it feels unresponsive

File: ext/action-popup.html [hunk 7]
```diff
@@ -11,94 +11,64 @@
...
-                <div class="nav-button-warning-badge-outer"></div>
-                <div class="nav-button-warning-badge-inner"></div>
+    <div>
+        <div tabindex="0" class="action-item action-select-profile" hidden>
+            <div class="action-item-inner">
+                <div class="action-item-left">
+                    <h2>Profile</h2>
+                </div>
+                <div class="action-item-right">
+                    <select class="profile-select" id="profile-select">
```
Comment (jamesmaa) [inline ext/action-popup.html:26]: This dropdown can only handle about 8 character strings before it cuts out and looks weird

image

File: ext/js/pages/action-popup-main.js (hunk 1 above)
Comment (jamesmaa) [reply ext/js/pages/action-popup-main.js:84]: Let's also have this tooltip display "No dictionary enabled" if there's no dictionary enabled

**Comment by Kuuuube** [comment]
Does this take into account the mobile version of the popup? It looks like youve just ripped out all of that.

File: ext/action-popup.html (hunk 3 above)
Comment (Kuuuube) [reply ext/action-popup.html:19]: Removing the loading message is an absolute no go. This needs to be brought back.

**Comment by MarvNC** [comment]
I like the colors you went with, with the gradient showing only when it's on.
Also CI is failing, do you have your local env set up correctly?

File: ext/action-popup.html (hunk 6 above)
Comment (Kuuuube) [reply ext/action-popup.html:47]: That's because theres an override on the animation duration here https://github.com/yomidevs/yomitan/blob/5f41352fd78cd14468e66e70007a39e2ee20f19c/ext/css/action-popup.css#L41

**Comment by koiyakiya** [comment]
image
Btw this might be confusing im not sure

**Comment by Kuuuube** [comment]
Theres a function somewhere that sets that text for the correct OS since it's always different. Can just borrow that.

**Comment by martholomew** [comment]
Screenshot_20241121_225458
Screenshot_20241121_225344




Screenshots of Firefox mobile and Chrome

File: ext/action-popup.html (hunk 7 above)
Comment (martholomew) [reply ext/action-popup.html]: I added an ellipsis so it doesnt run over the dropdown icon

**Review by jamesmaa — APPROVED** [review]
